<launch>
	<node name="keyboard" pkg="keyboard" type="keyboard" />

	<arg name="launch_prefix" default="" /> <!--set to value="gdbserver localhost:10000" for remote debugging-->
	<arg name="node_namespace" default="apriltag_ros_single_image_client_node" />
	<arg name="image_load_path" value="$(find apriltag_ros)/images/to_detect.jpg"/> <!-- Where to load image for analysis from -->
	<arg name="image_save_path" value="$(find apriltag_ros)/images/detected.jpg"/>

	<!-- Set parameters -->
	<rosparam command="load" file="$(find apriltag_ros)/config/settings.yaml" ns="$(arg node_namespace)" />
	<rosparam command="load" file="$(find apriltag_ros)/config/tags.yaml" ns="$(arg node_namespace)" />

        <node name="image_saver" pkg="image_view" type="image_saver" output="screen">
                <remap from="image" to="cv_camera/image_rect"/>
                <param name="save_all_image" value="false" />
                <param name="filename_format" value="$(find apriltag_ros)/images/to_detect.jpg" />
        </node>

	<node name="capture_apriltag" pkg="ieee_master" type="capture_apriltag" clear_params="true" output="screen">
	    <param name="image_load_path" type="str" value="$(arg image_load_path)" />
	    <param name="image_save_path" type="str" value="$(arg image_save_path)" />

	    <!-- Camera intrinsic parameters -->
	    <param name="fx" type="double" value="652.7934615847107" />
	    <param name="fy" type="double" value="653.9480389077635" />
	    <param name="cx" type="double" value="307.1288710375904" />
	    <param name="cy" type="double" value="258.7823279214385" />
	</node>
</launch>
